### PR TITLE
Tp 2299/claim passphrase backup

### PR DESCRIPTION
**Pull request by peterjah**
Please revew only last commit "feat: add badges:claim command"

### CODE REVIEW

**Comment by dlecan** [comment]
https://spacelephant.tpondemand.com/entity/2299

File: src/commands/resolve.ts [hunk 1]
```diff
@@ -34,8 +34,8 @@ export class ResolveCommand extends ReadCommand {
         return ResolveCommand;
     }
 
-    protected async do(flags: Record<string, any>, args?: Record<string, any>): Promise<any> {
-        const resolved: DidResolution<any> = await resolveUnikName(args?.did, flags);
+    protected async do(flags: Record<string, any>, args: Record<string, any>): Promise<any> {
+        const resolved: DidResolution<any> = await resolveUnikName(args.did, flags);
```
Comment (ftreguer) [inline src/commands/resolve.ts]: target or did argument ?

File: src/commands/send.ts [hunk 2]
```diff
@@ -70,7 +66,7 @@ export class SendCommand extends WriteCommand {
             return undefined;
         }
 
-        const recipientAddress = await cmdHelper.resolveWalletAddress(flags.to);
+        const recipientAddress = await cmdHelper.resolveWalletAddress(args.target);
```
Comment (ftreguer) [inline src/commands/send.ts]: targetResolve ?

File: src/commands/unik/disclose.ts [hunk 3]
```diff
@@ -11,21 +11,31 @@ import { cli } from "cli-ux";
...
-        ...unikidFlag("The UNIK token on which to disclose values"),
-        ...explicitValueFlag("Explicit value to disclose.", true),
-    };
+    public static flags = UnikDiscloseCommand.getFlags();
+    public static args = [getTargetArg()];
+
+    protected static getFlags() {
+        const flags = {
+            ...WriteCommand.getWriteCommandFlags(),
+            ...explicitValueFlag("Explicit value to disclose.", true),
```
Comment (ftreguer) [inline src/commands/unik/disclose.ts]: required? We don't use the unikname to get explicit to disclose ?

File: src/commands/unik/read.ts [hunk 4]
```diff
@@ -24,29 +23,33 @@ export class UnikReadCommand extends ReadCommand {
         return UnikReadCommand;
     }
 
-    protected async do(flags: Record<string, any>): Promise<NestedCommandOutput> {
-        checkUnikIdFormat(flags.unikid);
+    protected async do(flags: Record<string, any>, args: Record<string, any>): Promise<NestedCommandOutput> {
+        // let { unikid, ownerAddress, chainmeta, transactions } = await this.targetResolve(flags, args.target);
```
Comment (ftreguer) [inline src/commands/unik/read.ts]: To remove?

File: src/utils/command-utils.ts [hunk 5]
```diff
@@ -282,3 +282,18 @@ function getDidDocsUrl(): string {
 function getDocsUrl() {
     return "https://docs.uns.network";
 }
+
+export function getTargetArg() {
+    return {
+        name: "target",
+        description: `target unikid or @unikname (see ${getDidDocsUrl()})`,
```
Comment (ftreguer) [inline src/utils/command-utils.ts]: nit: capitalize first letter

File: test/__fixtures__/commands/delegate/vote.ts [hunk 6]
```diff
@@ -28,12 +28,12 @@ export const shouldExit = [
...
         args: [commandName, "-n", "dalinet"],
         errorMsg:
-            "» :stop: Command fail because of unexpected value for at least one parameter (id). Please check your parameters.;\n",
+            "» :stop: Command fail because of unexpected value for at least one parameter (target). Please check your parameters.;\n",
     },
     {
         description: "Should exit with code 1 if delegate id malformed (no @)",
         args: [commandName, "-n", "dalinet", "delegateId"],
-        errorMsg: "» :stop: Unikid parameter does not match expected format;\n",
+        errorMsg: '» :stop: Invalid argument "delegateId";\n',
```
Comment (ftreguer) [inline test/__fixtures__/commands/delegate/vote.ts]: Error message changes a lot, have you seen with @Juliette-Megret ?

File: src/commands/unik/disclose.ts [hunk 7]
```diff
@@ -31,7 +31,7 @@ export class UnikDiscloseCommand extends WriteCommand {
     protected static getFlags() {
         const flags = {
             ...WriteCommand.getWriteCommandFlags(),
-            ...explicitValueFlag("Explicit value to disclose.", true),
+            ...explicitValueFlag("Explicit value to disclose.", true, false),
```
Comment (ftreguer) [inline src/commands/unik/disclose.ts]: ok, forget my previous comment for the required part ;)

File: src/commands/unik/disclose.ts [hunk 8]
```diff
@@ -73,6 +73,15 @@ export class UnikDiscloseCommand extends WriteCommand {
     }
 
     protected async do(flags: Record<string, any>, args: Record<string, any>): Promise<any> {
+        if (!flags.explicitValue) {
+            if (isDid(args.target)) {
+                const explicitValue = args.target.split(":").pop();
+                flags.explicitValue = [explicitValue];
```
Comment (ftreguer) [inline src/commands/unik/disclose.ts]: not that good to change flags, it's input it could not be changed

File: package.json [hunk 9]
```diff
@@ -82,6 +82,9 @@
             },
             "delegate": {
                 "description": "Manage UNS delegates (`uns delegate` to display Delegate commands)"
+            },
+            "badges": {
+                "description": "Manage UNS badges (`uns badges` to display Delegate commands)"
```
Comment (ftreguer) [inline package.json:87]: `Badges commands`

File: src/commands/badges/claim.ts [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    protected getAvailableFormats(): Formater[] {
+        return [OUTPUT_FORMAT.json, OUTPUT_FORMAT.yaml];
+    }
+
+    protected getCommand(): typeof BaseCommand {
+        return BadgesClaimCommand;
+    }
+
+    protected getProperties(flags: Record<string, any>): { [_: string]: string } {
+        console.log(flags.badge);
```
Comment (ftreguer) [inline src/commands/badges/claim.ts]: to remove

File: src/commands/badges/claim.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        if (Object.keys(badges).includes(flags.badge)) {
+            console.log(badges[flags.badge]);
```
Comment (ftreguer) [inline src/commands/badges/claim.ts]: to remove

File: src/commands/badges/claim.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import { PropertiesUpdateCommand as UpdatePropertiesCommand } from "../../updatePropertiesCommand";
+import { getNetworksListListForDescription } from "../../utils";
+
+const BADGE_PREFIX = "Badges/";
+
+const badges: Record<string, any> = {
+    "passphrase-backup": {
+        coreStr: BADGE_PREFIX + "Security/Passphrase/Backup",
+    },
+};
```
Comment (ftreguer) [inline src/commands/badges/claim.ts:13]: We could mutualize in SDK another time

**Review by ftreguer — CHANGES_REQUESTED** [review]
Nice job, some changes required

File: src/commands/resolve.ts (hunk 1 above)
Comment (peterjah) [reply src/commands/resolve.ts]: Here its did, not target. We cannot pass a unikid. moreover its possible to request properties

File: src/commands/unik/disclose.ts (hunk 3 above)
Comment (peterjah) [reply src/commands/unik/disclose.ts]: here the "true" argument is for multiple !

**Review by ftreguer — APPROVED** [review]

**Pull request merged**